elastic/kibana
# [ResponseOps][Research] Make the timeout for ES requests match the rule type timeout

Resolves https://github.com/elastic/kibana-team/issues/722

## Summary

By default ES requests time out after 30s. This PR re-uses the task execution timeout, which can be configured per rule type, to configure the ES request `requestTimeout`.


### Checklist



### To verify

- Add the following to `kibana.yml `

```
xpack.alerting.rules.run.ruleTypeOverrides:
  - id: '.es-query'
    timeout: '30s' (any timeout of your choosing)
```
- Add a shard delay to queries types in the ES query rule:

```
    aggs: {
      delay: {
        shard_delay: {
          value: '45s', (any delay of your choosing)
        },
      },
    },
```
I was able to update the queries in the following files:
Query DSL: `x-pack/plugins/stack_alerts/common/build_sorted_events_query.ts`
KQL: `x-pack/plugins/triggers_actions_ui/common/data/lib/build_agg.ts`

Note: the way the KQL queries work it was difficult to get the request to timeout. I had to use really short timeouts and delays. Maybe there is a better way to test this, if anyone has ideas pls let me know :) 
- Start Kibana
- Create an ES query rule and verify that timeout is configurable, and that ES will throw an error if the request times out
